src/axislike.rs

```diff
@@ -688,3 +703,114 @@ impl From<DualAxisData> for Vec2 {
         data.xy
     }
 }
+
+/// The shape of the deadzone for a [`DualAxis`] input.
+///
+/// Input values that are on the line of the shape are counted as inside
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub enum DeadZoneShape {
+    /// Deadzone with the shape of a cross.
+    ///
+    /// The cross is represented by two rectangles. When using [`DeadZoneShape::Cross`],
+    /// make sure rect_1 and rect_2 do not have the same values, otherwise the shape will be a rectangle
+    Cross {
+        /// The width of the first rectangle.
+        rect_1_width: f32,
+        /// The height of the first rectangle.
+        rect_1_height: f32,
+        /// The width of the second rectangle.
+        rect_2_width: f32,
+        /// The height of the second rectangle.
+        rect_2_height: f32,
+    },
+    /// Deadzone with the shape of a rectangle.
+    Rect {
+        /// The width of the rectangle.
+        width: f32,
+        /// The height of the rectangle.
+        height: f32,
+    },
+    /// Deadzone with the shape of an ellipse.
+    Ellipse {
+        /// The horizontal radius of the ellipse.
+        radius_x: f32,
+        /// The vertical radius of the ellipse.
+        radius_y: f32,
+    },
+}
+
+impl Eq for DeadZoneShape {}
+impl std::hash::Hash for DeadZoneShape {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
```

Comment:
I have limited knowledge on how to implement `Hash`, so I just looked at `SingleAxis` and tried to go from there. Is this correct?

Reply:
This seems fine.

No clue about cargo fmt; I'm fine with leaving it a mystery.